ui/src/app/testing/fake-config-loader.service.ts

```diff
@@ -0,0 +1,15 @@
+import { ConfigLoaderService } from "../../environments/config-loader.service";
+
+export class FakeConfigLoaderService extends ConfigLoaderService {
+    private testEnvironmentConfig: object = {
+        apiUrl : "https://go.to.here.com"
```

Comment:
I don't have that much to add in other areas of this PR so I will nitpick the one detail I do know about... it is considered suboptimal practice to use valid domains because someone could register them and collect traffic sent to them. `example.com` is [defined by RFC](https://www.iana.org/domains/reserved) to be a safe target.

```suggestion
        apiUrl : "https://go.to.here.example.com"
```